pkg/volume/util/nestedpendingoperations/nestedpendingoperations.go

```diff
@@ -256,10 +257,21 @@ func (grm *nestedPendingOperations) isOperationExists(key operationKey) (bool, i
 			previousOp.key.nodeName == key.nodeName
 
 		if volumeNameMatch && podNameMatch && nodeNameMatch {
-			return true, previousOpIndex
+			// pending first
+			if previousOp.operationPending {
+				return true, previousOpIndex
+			}
+			// then exact match prior non-exact match
+			podNameExactMatch := previousOp.key.podName == key.podName
+			nodeNameExactMatch := previousOp.key.nodeName == key.nodeName
+			if opIndex == -1 || (podNameExactMatch && nodeNameExactMatch) {
+				opIndex = previousOpIndex
+			}
 		}
```

Comment:
Thanks for making the fix. Does following logic seems more direct?

```golang
        exactMatch := -1
	nonExactMatch := -1

	for previousOpIndex, previousOp := range grm.operations {
		volumeNameMatch := previousOp.key.volumeName == key.volumeName

		if volumeNameMatch {
			if key.podName == previousOp.key.podName && key.nodeName == previousOp.key.nodeName {
				exactMatch = previousOpIndex
				break
			}

			podNameMatch := previousOp.key.podName == EmptyUniquePodName ||
				key.podName == EmptyUniquePodName ||
				previousOp.key.podName == key.podName

			nodeNameMatch := previousOp.key.nodeName == EmptyNodeName ||
				key.nodeName == EmptyNodeName ||
				previousOp.key.nodeName == key.nodeName

			if podNameMatch && nodeNameMatch {
				nonExactMatch = previousOpIndex
			}
		}
	}
	if exactMatch != -1 {
		return true, exactMatch
	}
	if nonExactMatch != -1 {
		return true, nonExactMatch
	}
```

Basically we want to pick exactly matching operation over other ones.

Reply:
Well the code does not return first pending match though. One downside of current approach now is - we must iterate through all operations before `IsOperationExists` can return. This could potentially be expensive on a busy node.

Reply:
Yea, in worst case, there is no pending match operations at all but there is a non exact match at the header, we must still iterate all. But I think we don't change the time complexity of O(N) from global perspective, because we should always iterate the slice to find a match volume operation if there are many different volumes operations. shall we add a TODO to improve the algorithm using other data structure in future?

Reply:
Yeah I wonder if we can change the data structure used here from a slice to something that can accommodate these requirements in a better way. The code is also kinda hard to follow because of underlying complexity (not this PR's fault, it was always hard to follow).

Reply:
aggree,  need to refactor this data structure. 

proposing a kep to do this?